Suggestion for file_parsing documents

khoivan88 opened the pull request:
Hi,

I tried to leave some comments with each commit. I hope this might be helpful!

Best,

khoivan88 committed:
Propose change for 'backslash' vs 'forward slash'.

khoivan88 committed:
Add some spaces for indentation and suggest alternative way to access last element of a list

khoivan88 committed:
Suggest change to sample solution

1. Use of `elif` instead of `if` for more memory management efficiency
2. Use of context manager for opening and reading of file for better memory management using generator and also auto file closing
3. Use of `sum()` function
4. Change 'kcal/mole' to 'kcal/mol'

khoivan88 committed:
Suggest to fix line number and Add regex refs

The current example has `enumerate` list with 0-index. By default, humans do not read the first line as line 0. This suggestion should fix that.

janash commented on _episodes/02-file_parsing.md:
````diff
@@ -196,10 +204,17 @@ print(words)
 print(energy)
 ```
+
+We also recogize that "energy" is the last element of the list. Therefore, an alternative way to assign `energy` is:
````
Can you add this as a callout? (yellow boxed information) The callout would be about other ways to access list elements (negative indexing)

The syntax would be

> ## Title
> Text
> Text
{: .callout}

khoivan88 replied:
I have switched this block to callout format! I hope I did it correctly. I am not familiar with carpentries format!

janash replied:
Thank you! I have previewed it locally and this appears to be the correct format.

janash commented on _episodes/02-file_parsing.md:
```diff
@@ -237,48 +252,48 @@ energy = float(words[3])
+>>         if 'Electrostatics    ' in line:
+>>             electro_line = line
+>>             important_lines.append(electro_line)
+>>         elif 'Exchange       ' in line:
```
Use of `elif` here would require us to introduce it at an earlier point in the material. If it is not previously introduced and explained it may be confusing for beginners. I am in favor of this, but we would have to figure out where.

khoivan88 replied:
I agree, and new users would probably shouldn't need to care about how python run the program (continue running on `if` clause or skip, `else` if clause. Would you like me to switch back to `if` for this PR?

janash replied:
I like the idea of showing `elif` and `else`. Let's switch to `if` for this PR, but can you make an issue about introducing `elif` so we can consider where/if to add it?

Thanks for your attention to these lessons!

khoivan88 replied:
I have created issue #59.

janash commented on _episodes/02-file_parsing.md:
````diff
@@ -300,35 +315,37 @@ for linenum, line in enumerate(list_name):
 This block of code searches our file for the line that contains "Center" and reports the line number.
 ```
-for linenum, line in enumerate(data):
+for linenum, line in enumerate(data, start=1):
````
Cool trick! Didn't know about `start` in enumerate. If we include this it should be in a callout and not part of the main material.

janash replied:
Can you please remove `start` here and add this capability as a callout as well?

Typically for call outs it is material that we may or may not teach in person depending on the level of the group.

khoivan88 replied:
I have changed it back to the original together with other lines affected by this!

janash commented on _episodes/02-file_parsing.md:
```diff
@@ -345,6 +362,6 @@ Now we know that this is line 77 in our file (remember that you start counting a
 
 ## A final note about regular expressions
-Sometimes you will need to match something more complex than just a particular word or phrase in your output file.  Sometimes you will need to match a particular word, but only if it is found at the beginning of a line.  Or perhaps you will need to match a particular pattern of data, like a capital letter followed by a number, but you won't know the exact letter and number you are looking for.  T ...
+Sometimes you will need to match something more complex than just a particular word or phrase in your output file.  Sometimes you will need to match a particular word, but only if it is found at the beginning of a line.  Or perhaps you will need to match a particular pattern of data, like a capital letter followed by a number, but you won't know the exact letter and number you are looking for.  T ...
```
Thanks for the link suggestions!

khoivan88 replied:
Yeah, I first learn about regex at the Automate the Boring Stuff with Python book and I thought he did a very good job. I used the regex101 almost every time I need to test some new regex these days. I hope people might find them useful

khoivan88 committed:
Add python negative indexing as callout

I hope I did it correctly with the callout syntax

khoivan88 committed:
Replace `elif` clause with `if` clause

to avoid introducing too many new concepts. Introduce in this issue [#59](https://github.com/MolSSI-Education/python_scripting_cms/issues/59)

khoivan88 committed:
Change back to 0-index for `enumerate`

khoivan88 committed:
Remove extra period!

janash committed:
Update 02-file_parsing.md
